core/devmode/src/main/java/io/quarkus/dev/RuntimeUpdatesProcessor.java

```diff
@@ -126,7 +126,8 @@ public boolean doScan() throws IOException {
                             p -> pathToClassName(p),
                             p -> CopyUtils.readFileContentNoIOExceptions(p)));
         }
-        if (changedClasses.isEmpty() && !checkForConfigFileChange()) {
+        // No restart if there are no changed classes and no config file change
+        if (changedClasses.isEmpty() & !checkForConfigFileChange()) {
```

Comment:
@mkouba I think I wouldn't do it that way. I would make `scanForChangedClasses()` only scan for classes.

I would also make it return a boolean as we don't need the changed classes.

And I would have in `doScan()`:
```java
boolean changedClasses = checkForChangedClasses();
boolean configFileChanged = checkForConfigFileChange();

if (!changedClasses && ! configFileChanged) {
            return false;
}
```

I think it would be clearer than what we have now. Let me know what you think.

Reply:
At the very least I don't like the single '&' trick, I think it is a bit confusing.

Reply:
@gsmet You're right. This class would deserve some refactoring. I'm usually cautious about modifying code I'm not sure I fully understand ;-)

> At the very least I don't like the single '&' trick, I think it is a bit confusing.

@stuartwdouglas Well, it is confusing to perform some logic inside an expression at all. I just wanted to keep the changes minimal. And single `&` is not a trick but a regular operator with well-defined behavior ;-)

Anyway, I'll try to improve the PR a bit. Thanks!